optaplanner-core/src/main/java/org/optaplanner/core/impl/io/jaxb/GenericJaxbIO.java

```diff
@@ -0,0 +1,318 @@
+        SchemaFactory schemaFactory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+        Schema schema;
+        try {
+            schema = schemaFactory.newSchema(GenericJaxbIO.class.getResource(nonNullSchemaResource));
+        } catch (SAXException e) {
+            throw new IllegalArgumentException("Unable to read input schema resource (" + nonNullSchemaResource + ")", e);
+        }
+
+        Unmarshaller unmarshaller = createUnmarshaller();
+        unmarshaller.setSchema(schema);
+        ValidationEventStringCollector validationEventHandler = new ValidationEventStringCollector();
+
+        try {
+            unmarshaller.setEventHandler(validationEventHandler);
+        } catch (JAXBException jaxbException) {
+            String errMessage = String.format("Unable to set validation event handler to the unmarshaller for "
+                    + "a root element class (%s).", rootClass.getName());
+            throw new OptaPlannerXmlSerializationException(errMessage, jaxbException);
+        }
+
+        try {
+            return (T) unmarshaller.unmarshal(reader);
+        } catch (JAXBException jaxbException) {
+            String errMessage = String.format(ERR_MSG_READ, rootClass.getName());
+            if (validationEventHandler.hasEvents()) {
+                String errMessageWithValidationEvents = errMessage + "\n" + validationEventHandler.reportAll();
+                throw new OptaPlannerXmlSerializationException(errMessageWithValidationEvents, jaxbException);
+            } else {
+                throw new OptaPlannerXmlSerializationException(errMessage, jaxbException);
+            }
+        }
+    }
+
+    private Unmarshaller createUnmarshaller() {
+        try {
+            return jaxbContext.createUnmarshaller();
+        } catch (JAXBException e) {
+            String errMessage = String.format("Unable to create JAXB unmarshaller for a root element class (%s).",
+                    rootClass.getName());
+            throw new OptaPlannerXmlSerializationException(errMessage, e);
+        }
+    }
+
+    /**
+     * Reads the input XML using the {@link Reader} overriding elements namespaces. If an element already has a namespace and
+     * a {@link ElementNamespaceOverride} is defined for this element, its namespace is overridden. In case the element has no
+     * namespace, new namespace defined in the {@link ElementNamespaceOverride} is added.
+     * 
+     * @param reader input XML {@link Reader}; never null
+     * @param elementNamespaceOverrides never null
+     * @return deserialized object representation of the XML.
+     */
+    public T readOverridingNamespace(Reader reader, ElementNamespaceOverride... elementNamespaceOverrides) {
+        Objects.requireNonNull(reader);
+        Objects.requireNonNull(elementNamespaceOverrides);
+
+        final String errMessage = String.format("Unable to read the (%s) from XML with overriding elements' namespaces: %s.",
```

Comment:
error style, always use braces to mention values at runtime.
=> "... the rootClass (%s) ... namespaces (%s)."